src/main/java/net/dv8tion/jda/internal/handle/ApplicationCommandPermissionsUpdateHandler.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.dv8tion.jda.internal.handle;
+
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.events.interaction.command.ApplicationCommandUpdatePrivilegesEvent;
+import net.dv8tion.jda.api.events.interaction.command.ApplicationUpdatePrivilegesEvent;
+import net.dv8tion.jda.api.interactions.commands.privileges.IntegrationPrivilege;
+import net.dv8tion.jda.api.utils.data.DataArray;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.JDAImpl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ApplicationCommandPermissionsUpdateHandler extends SocketHandler
+{
+    public ApplicationCommandPermissionsUpdateHandler(JDAImpl api)
+    {
+        super(api);
+    }
+
+    @Override
+    protected Long handleInternally(DataObject content)
+    {
+        Guild guild;
+        if (!content.isNull("guild_id"))
+        {
+            long guildId = content.getUnsignedLong("guild_id");
+            guild = getJDA().getGuildById(guildId);
+            if (getJDA().getGuildSetupController().isLocked(guildId))
+                return guildId;
+            else if (guild == null)
+                return null;
```

Comment:
Do we need to emit a warning here about receiving updates for a guild we don't know about?

Reply:
Probably not a bad idea. I looked around and found what i think is a cache for replaying events when the guild was finally loaded? Do we want to add this too?

Reply:
likely not worth it

Reply:
No, we wont bother with cache replay here I don't think